patch-upstream.sh

```diff
@@ -22,22 +22,11 @@ cd opentelemetry-lambda/collector
 patch -p2 < ../../collector.patch
 
 # Replace OTel Collector with ADOT Collector
-go mod edit -replace github.com/open-telemetry/opentelemetry-lambda/collector/lambdacomponents=github.com/aws-observability/aws-otel-collector/pkg/lambdacomponents@v0.24.1
-
```

Comment:
Wondering what was the result in local for the following commands

```
./patch-upstream.sh
cd go/ 
GOARCH=amd64 ./build.sh
```

Reply:
```
bryaag@147ddac12c15 go % GOARCH=amd64 ./build.sh 
~/repos/aws-otel-lambda/opentelemetry-lambda/collector ~/repos/aws-otel-lambda/go
rm -rf build/
Building otel collector extension
mkdir -p /Users/bryaag/repos/aws-otel-lambda/opentelemetry-lambda/collector/build/extensions
GOOS=linux GO111MODULE=on CGO_ENABLED=0 installsuffix=cgo go build -trimpath -ldflags "-s -w -X main.GitHash=b52f9d506121d41799071e20a90c97ceb25da16e -X main.Version=v0.25.0 -X github.com/open-telemetry/opentelemetry-collector-contrib/exporter/awsxrayexporter.collectorDistribution=opentelemetry-collector-lambda" -o /Users/bryaag/repos/aws-otel-lambda/opentelemetry-lambda/collector/build/extensions .
Package zip file for collector extension layer
mkdir -p /Users/bryaag/repos/aws-otel-lambda/opentelemetry-lambda/collector/build/collector-config
cp config* /Users/bryaag/repos/aws-otel-lambda/opentelemetry-lambda/collector/build/collector-config
cd /Users/bryaag/repos/aws-otel-lambda/opentelemetry-lambda/collector/build && zip -r collector-extension.zip collector-config extensions
  adding: collector-config/ (stored 0%)
  adding: collector-config/config.yaml (deflated 52%)
  adding: extensions/ (stored 0%)
  adding: extensions/collector (deflated 68%)
~/repos/aws-otel-lambda/go
go: downloading go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp v0.27.0
go: downloading go.opentelemetry.io/otel v1.2.0
go: downloading go.opentelemetry.io/contrib/instrumentation/github.com/aws/aws-lambda-go/otellambda v0.27.0
go: downloading go.opentelemetry.io/contrib/instrumentation/github.com/aws/aws-lambda-go/otellambda/xrayconfig v0.27.0
go: downloading go.opentelemetry.io/contrib/instrumentation/github.com/aws/aws-sdk-go-v2/otelaws v0.27.0
go: downloading github.com/aws/aws-sdk-go-v2/service/s3 v1.11.0
go: downloading go.opentelemetry.io/contrib/propagators/aws v1.2.0
go: downloading github.com/aws/aws-sdk-go-v2/config v1.4.1
go: downloading github.com/aws/aws-lambda-go v1.27.0
go: downloading go.opentelemetry.io/otel/trace v1.2.0
go: downloading go.opentelemetry.io/otel/sdk v1.2.0
go: downloading github.com/felixge/httpsnoop v1.0.2
go: downloading go.opentelemetry.io/otel/metric v0.25.0
go: downloading go.opentelemetry.io/contrib/detectors/aws/lambda v0.27.0
go: downloading go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc v1.2.0
go: downloading go.opentelemetry.io/otel/exporters/otlp/otlptrace v1.2.0
go: downloading github.com/aws/aws-sdk-go-v2 v1.11.0
go: downloading github.com/aws/smithy-go v1.9.0
go: downloading golang.org/x/sys v0.0.0-20210423185535-09eb48e85fd7
go: downloading go.opentelemetry.io/otel/internal/metric v0.25.0
go: downloading google.golang.org/grpc v1.42.0
go: downloading google.golang.org/genproto v0.0.0-20200526211855-cb27e3aa2013
go: downloading go.opentelemetry.io/proto/otlp v0.11.0
go: downloading github.com/cenkalti/backoff/v4 v4.1.1
go: downloading golang.org/x/text v0.3.0
go: downloading github.com/aws/aws-sdk-go-v2/credentials v1.3.0
go: downloading github.com/aws/aws-sdk-go-v2/service/internal/s3shared v1.5.0
go: downloading github.com/aws/aws-sdk-go-v2/service/internal/accept-encoding v1.2.0
go: downloading github.com/aws/aws-sdk-go-v2/service/internal/presigned-url v1.2.0
go: downloading github.com/aws/aws-sdk-go-v2/feature/ec2/imds v1.2.0
go: downloading github.com/aws/aws-sdk-go-v2/internal/ini v1.1.0
go: downloading github.com/aws/aws-sdk-go-v2/service/sso v1.3.0
go: downloading github.com/aws/aws-sdk-go-v2/service/sts v1.5.0
  adding: bootstrap (deflated 49%)
```

Reply:
Output from the patch
```
bryaag@147ddac12c15 aws-otel-lambda % ./patch-upstream.sh                                                                                                                                                                   
patching file collector.go                                                                                                                                                                                                  
Hunk #1 succeeded at 19 with fuzz 2.                                                                                                                                                                                        
Hunk #2 succeeded at 61 (offset 2 lines).
Hunk #3 succeeded at 92 with fuzz 2 (offset 3 lines).
go: downloading github.com/aws-observability/aws-otel-collector/pkg/lambdacomponents v0.25.1
bryaag@147ddac12c15 aws-otel-lambda % cd go 
```